src/Compiler/FSComp.txt

```diff
@@ -679,7 +679,7 @@ tcUnnamedArgumentsDoNotFormPrefix,"The unnamed arguments do not form a prefix of
 824,tcAttributesAreNotPermittedOnLetBindings,"Attributes are not permitted on 'let' bindings in expressions"
 825,tcDefaultValueAttributeRequiresVal,"The 'DefaultValue' attribute may only be used on 'val' declarations"
 826,tcConditionalAttributeRequiresMembers,"The 'ConditionalAttribute' attribute may only be used on members"
-827,tcInvalidActivePatternName,"This is not a valid name for an active pattern"
+827,tcInvalidActivePatternName,"'%s' is not a valid method name. Use a 'let' binding instead."
```

Comment:
Regarding this particular wording change, why and where do you think it makes things clearer?

The reason I am asking is that I am thinking about beginners and intermediate F# programmers and such a wording might be not very helpful for them. "Binding" might be a hard word really, for a long time I was reading it like "thing". As in, "use let _binding_ here" would be just "use some let _thing_ here" for me. And in cases like `let (|O|K|_|) = None`, I am already using some "let _thing_" but you're telling me to use it anyway. 

Also, you're telling me this is a bad method name, but I didn't mean this as a method. I really just mistyped `let (|OK|_|) = None`.

What I am saying is - although it's generally good to be precise and technically correct with diags - sometimes it actually can be better to just say "this is an invalid identifier" without further advice, especially when a typo or a feature misuse is likely.

Might not be the case here (hence the question) but in such situations the new error you're adding IMO makes a good error combo with the one that exists currently.

Reply:
Yeah. As we know have a new dedicated error message for multi-case partial active patterns(not supported) `tcInvalidActivePatternName` will only now only be used in cases where active patterns are interpreted by the plain Long Idents e.g when used as `static members`:

![Screenshot 2024-09-02 at 17 26 14](https://github.com/user-attachments/assets/1cb2a202-26f9-490c-b7b4-c7de83c6e31f)


```fsharp
type APUsageInClass() =
    static member (|A|) = 7
    static member (|C|_|) = 5 
    static member (|A|B|) x = A // '(|A|B|)' is not a valid method name. Use a 'let' binding instead.

module Tests =
    APUsageInClass.(|A|)
    APUsageInClass.(|A|) + APUsageInClass.(|C|_|)
    APUsageInClass.(|C|_|)
    APUsageInClass.(|A|B|) 5
    let ss x =     
        match x with
        | APUsageInClass.``(|A|)`` -> printfn "A"
        | APUsageInClass.``(|A|B|)`` -> printfn "AB"
        | APUsageInClass.``(|C|_|)`` -> printfn "C"
        |  _ -> printf "Sad like a panda"
```

Reply:
Yeah that's better probably :) maybe @vzarytovskii you have an opinion since you're here, or @brianrourkeboll as an active active pattern user

Reply:
So I think this is just some whitespace crap, I pushed the changes, things pass for me at least locally.

Reply:
We should probably trim lines when comparing results to baselines, we do it in other test suites.

Reply:
Thanks for looking into this. Looks like like there are some other unrelated changes in `SpanTests.fs`.  Wil investigate tomorrow.

Reply:
Damn I commented in the wrong thread. No that was unrelated, caused by the broken main we had for a bit. It should be all good now after branch update